src/Context.php

```diff
@@ -14,15 +14,111 @@ class Context extends \ArrayObject
     public string $currentDirectory;
 
     /**
-     * @param array<(int|string), TValue> $array The input parameter accepts an array or an Object
+     * @param array<(int|string), TValue> $data The input parameter accepts an array or an Object
      * @param array<string, string> $environment a list of environment variables to add to the command
      */
     public function __construct(
-        public array $array = [],
+        array $data = [],
```

Comment:
don't we want to work only with immutable state ? if yes then we could have everything in readonly (imo it's the best way for context to avoid understanding issue)